### ISSUE TITLE
STDSChallengeResponseViewController not respecting preferredStatusBarStyle setting.

### ISSUE THREAD

**Issue by adamcarter93** [stripe/stripe-ios#1301]
## Summary
When presenting a STDSChallengeResponseViewController, the status bar colour is not changing to the preferredStatusBarStyle setting. i.e. when set to lightContent it shows a dark status bar.

## Code to reproduce 
    let handler = STPPaymentHandler.shared()
    handler.threeDSCustomizationSettings = STPThreeDSCustomizationSettings.default()
    handler.threeDSCustomizationSettings.uiCustomization = STPThreeDSUICustomization.defaultSettings()
    handler.threeDSCustomizationSettings.uiCustomization.preferredStatusBarStyle = .lightContent

## iOS version
iOS 12.4 (16G73)

## Installation method
Carthage

## SDK version
16.0.6

**Comment by yuki-stripe** [comment]
Hi @adamcarter93,

From my testing, it looks like the view controller's `preferredStatusBarStyle` is set to the expected value.  However, it also looks like the value is ignored, and the status bar text color is black or white depending on the [navigation bar style](https://stripe.dev/stripe-ios/docs/Classes/STPThreeDSNavigationBarCustomization.html#/c:objc(cs)STPThreeDSNavigationBarCustomization(py)barStyle).  This is with setting `UIViewControllerBasedStatusBarAppearance` to `YES` in Info.plist.

See https://stackoverflow.com/questions/19022210/preferredstatusbarstyle-isnt-called/19513714#19513714

In short, does setting `barStyle = .black` work for you?  Setting `barTintColor` might let you still control the actual color of the navigation bar, I'm not sure.

**Comment by yuki-stripe** [comment]
I've removed `preferredStatusBarStyle` since it doesn't actually do anything in our case (because our UIViewControllers are in a UINavigationController).  Instead, set the [barStyle](https://stripe.dev/stripe-ios/docs/Classes/STPThreeDSNavigationBarCustomization.html#/c:objc(cs)STPThreeDSNavigationBarCustomization(py)barStyle) to `.black` to set the statusBarStyle to `.lightContent`.  

Check out the summary here https://github.com/stripe/stripe-ios/pull/1308
